tests/v2/test_waku_rln_relay.nim

```diff
@@ -720,3 +721,41 @@ suite "Waku rln relay":
     # check that the conversion has not distorted the original value
     check:
       keypair.get().idCommitment == idCommitment
+
+  test "Read Persistent RLN credentials":
+    # create an RLN instance
+    var rlnInstance = createRLNInstance()
+    check:
+      rlnInstance.isOk == true
+
+    var key = membershipKeyGen(rlnInstance.value)
+    var empty: array[32, byte]
+    check:
+      key.isSome
+      key.get().idKey.len == 32
+      key.get().idCommitment.len == 32
+      key.get().idKey != empty
+      key.get().idCommitment != empty
+
+    debug "the generated membership key pair: ", key
+
+    var 
+      k: MembershipKeyPair
+      index: MembershipIndex
+
+    index = MembershipIndex(1)
+    k = key.get()
+
+    var rlnMembershipCredentials = RlnMembershipCredentials(membershipKeyPair: k, rlnIndex: index)
+
+    let path = "testPath.txt"
+
+    # Write RLN credentials
+    writeFile(path, pretty(%rlnMembershipCredentials))
+
+    var credentials = readPersistentRlnCredentials(path)
+
+    check:
+      credentials.membershipKeyPair == k
+    
```

Comment:
You should check the index too

Reply:
Refer 51166dbcb9f46b185381ee0130ec075a69f525e5